### PR TITLE
Please test this!

### PR DESCRIPTION
**Pull request by robske110**
UniversalJoin

### CODE REVIEW

**Comment by robske110** [comment]
@aodzip

**Comment by udachin26** [comment]
Can I help?
I have a VDS.

**Comment by robske110** [comment]
You can download the zip of this branch and the run it and see if it's working

**Comment by udachin26** [comment]
Well now all set!

**Comment by udachin26** [comment]
Perhaps I am a newbie, but where to download ZIP kernel?

**Comment by robske110** [comment]
Here https://github.com/robske110/ClearSky/tree/patch-50 click download zip

**Comment by udachin26** [comment]
ClearSky Crash Dump Thu Apr 28 15:19:01 MSK 2016

Error: Call to a member function toBinary() on null
File: /src/pocketmine/Player
Line: 2228
Type: E_ERROR

Code:
[2219]                      $this->directDataPacket($pk);
[2220]                  }
[2221]                  $this->close("", $message, false);
[2222]                  break;
[2223]              }
[2224]              $this->randomClientId = $packet->clientId;
[2225]              $this->loginData = ["clientId" => $packet->clientId, "loginData" => null];
[2226] 
[2227]              $this->uuid = $packet->clientUUID;
[2228]              $this->rawUUID = $this->uuid->toBinary();
[2229]              $this->clientSecret = $packet->clientSecret;
[2230]              $valid = true;
[2231]              $len = strlen($packet->username);
[2232]              if($len > 16 or $len < 3){
[2233]                  $valid = false;
[2234]              }
[2235]              for($i = 0;$i < $len and $valid;++$i){
[2236]                  $c = ord($packet->username{$i});
[2237]                  if(($c >= ord("a") and $c <= ord("z")) or
[2238]($c >= ord%28"A"%29 and $c <= ord%28"Z"%29) or

Backtrace:
#0 (): pocketmine\Server->crashDump(boolean)

ClearSky version: 1.1 #CuttingEdge [Protocol 60; API 1.13.1]
uname -a: Linux medium-world.ru 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt20-1+deb8u3 (2016-01-17) x86_64
PHP Version: 5.6.10
Zend version: 2.6.0
OS : Linux, linux

Loaded plugins:
AntiPiar 1 by TheMaks for API(s) 1.12.0
ServerAuth 2.13 by EvolSoft for API(s) 1.11.0
RegionGuard 0.0.1 by Skaleta for API(s) 1.6.0
PurePerms 1.3.7 by 64FF00 & ProjectInfinity for API(s) 1.13.0, 2.0.0
VIPSlots 0.0.4-1 by iJoshuaHD for API(s) 1.12.0, 1.13.0, 1.13.1, 2.0.0
EconomyAPI 2.0.9 by onebone for API(s) 1.0.0, 1.1.0, 1.2.0, 1.2.1, 1.3.0, 1.3.1, 1.4.0, 1.4.1
SkyWars 5.0.0 by Bruno for API(s) 1.0.0, 1.1.0, 2.0.0
TAGBlock 0.7-Alpha by ifteam for API(s) 1.13.0, 2.0.0

**Comment by udachin26** [comment]
I tried to go to 14.1, but the server will stop!

File: src/pocketmine/network/protocol/Info.php [hunk 1]
```diff
@@ -11,8 +11,8 @@
 	 * Actual Minecraft: PE protocol versions
 	 */
 	const CURRENT_PROTOCOL = 60;
-	const ACCEPT_PROTOCOL = [60];
-	//This is for beta 1-bla and 0.14.0-0.14.bla , I think no need to change this - Aodzip
+	const ACCEPT_PROTOCOL = [40,41,42,44,45,46,47,60];
+	//CURRENT_PROTOCOL M U S T BE THE LOWEST NUMBER IIN ACCEPT_PROTOCOL ARRAY! (CurrentProtocol does not have to be the same as in pocketmine.yml)
```
Comment (inxomnyaa) [inline src/pocketmine/network/protocol/Info.php]: *IN
And it mustn't. Guess why it worked before

**Comment by udachin26** [comment]
https://github.com/iTXTech/Genisys/commit/eded9311899ce69eb60259e47816b8a79a0a3599
What?

**Comment by robske110** [comment]
Try it now, this should work

**Comment by inxomnyaa** [comment]
Still getting the freezes

File: src/pocketmine/resources/pocketmine.yml [hunk 2]
```diff
@@ -111,7 +111,7 @@ network:
  #Interrupt Raklib after reach timeout, save CPU usage, may cause packet lost.(-1 for disable , all pkt will send)
  protocol: 60
```
Comment (SalmonDE) [inline src/pocketmine/resources/pocketmine.yml:112]: Don't you also need to change this?

**Comment by udachin26** [comment]
@robske110
Hello, can I speak with you anywhere, in any social network?
Twitere? VKontakte? FaceBook? Skype?

**Comment by robske110** [comment]
Gitter would be possible
Twitter is @robskeBueba

**Pull request closed without merging**